core/src/main/java/org/apache/druid/timeline/partition/SingleDimensionPartialShardSpec.java

```diff
@@ -94,14 +125,17 @@ public boolean equals(Object o)
     if (o == null || getClass() != o.getClass()) {
       return false;
     }
-    SingleDimensionShardSpecFactory that = (SingleDimensionShardSpecFactory) o;
-    return Objects.equals(partitionDimension, that.partitionDimension) &&
-           Objects.equals(partitionBoundaries, that.partitionBoundaries);
+    SingleDimensionPartialShardSpec that = (SingleDimensionPartialShardSpec) o;
```

Comment:
May be useful to add an `EqualsVerifier` test for this.

Reply:
Added.